src/hera/workflows/_mixins.py

```diff
@@ -424,8 +424,7 @@ def __call__(self, *args, **kwargs) -> SubNodeMixin:
             return Task(*args, template=self, **kwargs)
         except InvalidType:
             pass
-
-        raise InvalidTemplateCall("Container is not under a Steps, Parallel, or DAG context")
+        return None
```

Comment:
@flaviuvadan curious around the reasoning behind this change. Could you provide some color?

Reply:
Could we still raise the exception and catch it in the right place where it's expected?

Reply:
This was leftover from testing, readded